simple_app/app_simple.py

```diff
@@ -148,14 +156,66 @@ def all_sources():
     return allresults
 
 
+def find_colours(photodf: pd.DataFrame, allbands: np.ndarray):
+    """
+    Find all the colours using available photometry
+
+    Parameters
+    ----------
+    photodf: pd.DataFrame
+        The dataframe with all photometry in
+    allbands: np.ndarray
+        All the photometric bands
+
+    Returns
+    -------
+    photodf: pd.DataFrame
+        The dataframe with all photometry and colours in
+    """
+    for i, band in enumerate(allbands):  # loop over all bands TODO: sort by wavelength?
+        if i + 1 == len(allbands):  # last band
+            break
+        nextband: str = allbands[i + 1]  # next band
+        photodf[f'{band}_{nextband}'] = photodf[band] - photodf[nextband]  # colour
+    return photodf
+
+
+def all_photometry():
+    """
+    Get all the photometric data from the database to be used in later CMD as background
+
+    Returns
+    -------
+    allphoto: pd.DataFrame
+        All the photometry in a dataframe
+    allbands: np.ndarray
+        The unique passbands to create dropdowns by
+    """
+    db = SimpleDB(db_file, connection_arguments={'check_same_thread': False})  # open database
+    allphoto: pd.DataFrame = db.query(db.Photometry).pandas()  # get all photometry
+    allbands: np.ndarray = allphoto.band.unique()  # the unique bands
+    outphoto = pd.DataFrame({band: np.empty(len(allphoto.source.unique())) for band in allbands})  # initial dict
+    i = 0  # start counter
+    # FIXME: some sources have multiple mags for the same band
```

Comment:
Yea there are several ways to handle grouping similar objects, including doing so with the DB query itself rather than pandas. Make an issue for this and we can work on it for a future pull request.

Reply:
Handled